api/base/permissions.py

```diff
@@ -53,6 +54,12 @@ def _get_scopes(self, request, view):
             except AttributeError:
                 raise ImproperlyConfigured('TokenHasScope requires the view to define the '
                                            'required_read_scopes attribute')
+            assert is_iterable_but_not_string(view.required_read_scopes), \
+                'The required_read_scopes must be an iterable of CoreScopes'
+            if view.required_read_scopes and isinstance(view.required_read_scopes[0], tuple):
```

Comment:
Unless I'm missing something, wouldn't it be better to check each of the required_read_scopes items and verified `isinstance(scope, CoreScope)`? Sorry I missed the earlier discussion. That seems the most direct way of determining if everything is correct, yes?